jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/AbstractConnection.java

```diff
@@ -90,25 +83,32 @@ protected Executor getExecutor()
     }
 
     /**
-     * <p>Utility method to be called to register read interest.</p>
-     * <p>After a call to this method, {@link #onFillable()} or {@link #onFillInterestedFailed(Throwable)}
-     * will be called back as appropriate.</p>
+     * <p>Registers read interest using the default {@link Callback} with {@link Invocable.InvocationType#BLOCKING}.</p>
+     * <p>When read readiness is signaled, {@link #onFillable()} or {@link #onFillInterestedFailed(Throwable)}
+     * will be invoked.</p>
+     * <p>This method should be used sparingly, mainly from {@link #onOpen()}, and {@link #fillInterested(Callback)}
+     * should be preferred instead, passing a {@link Callback} that specifies the {@link Invocable.InvocationType}
+     * for each specific case where read interest needs to be registered.</p>
      *
+     * @see #fillInterested(Callback)
      * @see #onFillable()
+     * @see #onFillInterestedFailed(Throwable)
      */
     public void fillInterested()
     {
-        if (LOG.isDebugEnabled())
-            LOG.debug("fillInterested {}", this);
-        getEndPoint().fillInterested(_readCallback);
+        fillInterested(_readCallback);
+    }
+
+    public void nonBlockingFillInterested()
```

Comment:
javadoc

Reply:
I'm wondering if rather than having different `fillInterested` methods, would it be clearer to have `onFillable` and `onFillableNonBlocking`.  If `onFillable` is extended, then the instance is BLOCKING, otherwise it is none blocking?

Reply:
or better yet, just have `AbstractConnection.NonBlocking` ?    or do we need to mix/match blocking and non blocking demand?  Surely not as onFillable implementation is fixed?

Reply:
@sbordet I don't understand.  The onFillable method is either blocking or non-blocking.  Or are you saying it is stateful and can sometimes be either?

We always have `fillInterested(Callback)` if we want an alternative invocation type.       

I think it is far better to have a fixed known invocation type for the onFillable method and use that for `fillInterested` rather than risk calling the wrong `fillInterested` method when the `onFillable` is in a different state.

Reply:
Done, introduced `AbstractConnection.NonBlocking`.